cwms-data-api/src/main/java/cwms/cda/api/project/ProjectLockRequest.java

```
@@ -0,0 +1,147 @@
+            },
+            responses = {
+                @OpenApiResponse(status = STATUS_200, content = {
+                    @OpenApiContent(type = Formats.JSON, from = ProjectLockId.class)}
+                )},
+            method = HttpMethod.POST,
+            path = PATH,
+            tags = {TAGS}
+    )
+    @Override
+    public void handle(@NotNull Context ctx) throws Exception {
+
+        try (Timer.Context ignored = markAndTime(GET_ALL)) {
+            ProjectLockDao lockDao = new ProjectLockDao(JooqDao.getDslContext(ctx));
+
+            String reqContentType = ctx.req.getContentType();
+            String formatHeader = reqContentType != null ? reqContentType : Formats.JSON;
+            ContentType contentType = Formats.parseHeader(formatHeader, ProjectLock.class);
+            ProjectLock lock = Formats.parseContent(contentType, ctx.bodyAsInputStream(), ProjectLock.class);
+            boolean revokeExisting =
+                    ctx.queryParamAsClass(REVOKE_EXISTING, Boolean.class).getOrDefault(REVOKE_DEFAULT);
+            int revokeTimeout =
+                    ctx.queryParamAsClass(REVOKE_TIMEOUT, Integer.class).getOrDefault(DEFAULT_TIMEOUT);
+
+            String lockId = lockDao.requestLock(lock, revokeExisting, revokeTimeout);
+            if (lockId != null) {
+                ProjectLockId id = new ProjectLockId(lockId);
+                String acceptHeader = ctx.header(Header.ACCEPT);
+                ContentType responseType = Formats.parseHeader(acceptHeader, ProjectLockId.class);
+                String result = Formats.format(responseType, id);
+                ctx.result(result);
+                ctx.contentType(responseType.toString());
+                requestResultSize.update(result.length());
+                ctx.status(HttpServletResponse.SC_CREATED);
+            } else {
+                // We don't have any idea exactly why the create failed - right?
```

Create fail doesn't throw an error? Probably needs a Jira ticket to correct that behavior

---

It looks like the pl/sql does throw exceptions.

---

so the `lockId` won't ever come back as null?

---

No, I didn't say that. pl/sql looks like it does return null if it thinks it got into a race condition.

---

The database should throw an exception if that happens so we should create a db ticket.